### PR TITLE
feat: Upgrading edx-drf-extensions and edx-auth-backends ( using pyjwt=2.1.0 )

### PR DESCRIPTION
**Pull request by awais786**
This Upgrade PR contains breaking changes due to pyjwt==2.1.0 in edx-drf-extensions and edx-auth-backend latest versions.

NOTE: This PR needs proper testing on stage. If gocd shows some previous un-deployed commit please deploy them first and deploy this later to minimize scope.

JIRA: https://openedx.atlassian.net/browse/BOM-2758

### CODE REVIEW

**Comment by awais786** [comment]
https://github.com/edx/edx-analytics-data-api/pull/470
similar PR.

File: registrar/apps/api/tests/mixins.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def default_payload(self, user, admin=False, ttl=5):
 
     def generate_jwt_header(self, user, admin=False, ttl=5, **overrides):
         """Generate a jwt header value for AUTHORIZATION"""
-        jwt_token = self.generate_id_token(user, admin, ttl, **overrides).decode('utf-8')
```
Comment (awais786) [inline registrar/apps/api/tests/mixins.py:135]: https://pyjwt.readthedocs.io/en/stable/changelog.html#jwt-encode-return-type

File: requirements/common_constraints.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+    # This is a temporary solution to override the real common_constraints.txt
```
Comment (iamsobanjaved) [inline requirements/common_constraints.txt:1]: Fix space issue here by removing space from here: https://github.com/edx/registrar/pull/433/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R48

**Review by iamsobanjaved — APPROVED** [review]
Just a minor improvement in a comment.

**Comment by iamsobanjaved** [comment]
@schenedx can you please review this PR. This needs special attention before deployment and testing on the stage of specifically API endpoints with JWT authentication enabled. Similar PRs in credentials, analytics-api, and course-discovery already deployed on production.

**Review by schenedx — APPROVED** [review]
LGTM. 
The test case we can use when this deploys to stage is to hit https://program-console.stage.edx.org. Make sure the list of programs can load on the page.
Then interact with the app by downloading enrollments, and uploading enrollments.

**Comment by iamsobanjaved** [comment]
@schenedx can you please ask someone from your team to handle this deployment?

**Comment by schenedx** [comment]
I can handle this myself. I don't feel comfortable with this on a Friday. I'll do this next week.

**Comment by matthugs** [comment]
@iamsobanjaved could you link to a ticket which captures this work?

**Comment by iamsobanjaved** [comment]
Sure, this is the ticket for this upgrade: https://openedx.atlassian.net/browse/BOM-2758

Updated the description as well.

**Pull request merged**